Source/WebCore/rendering/RenderBlock.cpp

```diff
@@ -1120,6 +1129,11 @@ void RenderBlock::paint(PaintInfo& paintInfo, const LayoutPoint& paintOffset)
     if (pushedClip)
         popContentsClip(paintInfo, phase, adjustedPaintOffset);
 
+    if (paintInfo.phase == PaintPhase::Foreground && !renderCaretInsideContentsClip()) {
+        paintCaret(paintInfo, adjustedPaintOffset, CursorCaret);
+        paintCaret(paintInfo, adjustedPaintOffset, DragCaret);
```

Comment:
Can we factor these two lines into a separate method, since it's duplicated below?

Reply:
Good idea, I'll make that change